.github/workflows/benchmark.yml

```diff
@@ -16,31 +19,32 @@ jobs:
       - name: Check out the repository
         uses: actions/checkout@v4
         with:
-          lfs: true
+          fetch-depth: 0
 
       - uses: tibdex/github-app-token@v1
         id: generate-token
         with:
           app_id: ${{ secrets.APP_ID }}
           private_key: ${{ secrets.APP_PRIVATE_KEY }}
 
-      # Use cached python and dependencies, install poetry
-      - name: "Setup Python, Poetry and Dependencies"
-        uses: packetcoders/action-setup-cache-python-poetry@main
+      # Setup python
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v5
         with:
-          python-version: 3.9
-          poetry-version: 1.8.3
+          python-version: "3.10"
 
-      - name: Install project dependencies
-        run: poetry install
+      - name: Install dependencies
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
 
-      - name: Install project dependencies
-        run: poetry install
+      - name: Install package
```

Comment:
I saw that there's already an official action by uv
`uses: astral-sh/setup-uv@v5`

Reply:
https://docs.astral.sh/uv/guides/integration/github/#installation

Reply:
I tested this, but did not see any advantage. On the contrary, it tries to use some kind of cache, this however takes longer than without, and also gets confused during the matrix tests. I'll revert to the previous situation.

Reply:
Matrix method requires (from the link)

```
      - name: Install uv and set the python version
        uses: astral-sh/setup-uv@v5
        with:
          python-version: ${{ matrix.python-version }}
```

Cache should generally be faster after first time, no?

Reply:
The caches it created are only 8mbyte or so, so they don't contain anything important. UV hard-links the packages to versions available already on the system, thus the installation is super-fast, the whole setup takes less then 10s. Using caches, I am not sure we will be using the most-recent package version all the time, thus I prefer the current workflow. I run the multiversion tests now every week to see that thinks keep working if packages update. This way, we don't have to constantly update a lockfile (which you can also use with uv, but I don't see the real advantage for our purposes).